cmd/installer/cmd/image.go

```diff
@@ -0,0 +1,391 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package cmd
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"html/template"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/talos-systems/talos/cmd/installer/pkg"
+	"github.com/talos-systems/talos/internal/pkg/installer"
+	"github.com/talos-systems/talos/internal/pkg/kernel"
+	"github.com/talos-systems/talos/internal/pkg/runtime"
+	"github.com/talos-systems/talos/internal/pkg/runtime/platform"
+	"github.com/talos-systems/talos/pkg/cmd"
+	"github.com/talos-systems/talos/pkg/config/types/v1alpha1"
+	"github.com/talos-systems/talos/pkg/constants"
+	"github.com/talos-systems/talos/pkg/version"
+)
+
+var (
+	configArg       string
+	bootloaderArg   bool
+	diskArg         string
+	platformArg     string
+	outputArg       string
+	extraKernelArgs []string
+)
+
+// imageCmd represents the image command
+var imageCmd = &cobra.Command{
+	Use:   "image",
+	Short: "",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
+		if err := runImageCmd(); err != nil {
+			log.Fatal(err)
+		}
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(imageCmd)
+
+	imageCmd.Flags().StringVar(&configArg, "config", "none", "The value of "+constants.KernelParamConfig)
+	imageCmd.Flags().BoolVar(&bootloaderArg, "bootloader", true, "Install a booloader to the specified disk")
+	imageCmd.Flags().StringVar(&diskArg, "disk", "", "The path to the disk to install to")
+	imageCmd.Flags().StringVar(&platformArg, "platform", "", "The value of "+constants.KernelParamPlatform)
+	imageCmd.Flags().StringArrayVar(&extraKernelArgs, "extra-kernel-arg", []string{}, "Extra argument to pass to the kernel")
```

Comment:
This should probably be persistent flags on `root` command since `image` and `install` will both use them.

Reply:
On second thought, maybe not... only `platform` makes sense here.